templates/common/baremetal/files/baremetal-keepalived.yaml

```diff
@@ -57,29 +34,79 @@ contents:
           - name: NSS_SDB_USE_CACHE
             value: "no"
         command:
-        - /usr/sbin/keepalived
-        args:
-        - "-f"
-        - "/etc/keepalived/keepalived.conf"
-        - "--dont-fork"
-        - "--vrrp"
-        - "--log-detail"
-        - "--log-console"
+        - /bin/bash
+        - -c
+        - |
+          #/bin/bash
+          reload_keepalived()
+          {
+            declare -r pid="$(pgrep -o keepalived)"
+            kill -s SIGHUP "$pid"
+          }
+
+          msg_handler()
+          {
+            while read -r line; do
+              echo "The client sent: $line"  >&2
+              # currently only 'reload' msg is supported
+              if [ "$line" = reload ]; then
+                  reload_keepalived
+              fi
+            done
+          }
+
+          set -ex
+          declare -r keepalived_sock="/var/run/keepalived/keepalived.sock"
+          export -f msg_handler
+          export -f reload_keepalived
+          if [ -s "/etc/keepalived/keepalived.conf" ]; then
```

Comment:
There is a regression. Previously it would fail if the config does not exist.

Reply:
Now instead of failing it will just wait in socat for a reload when the config is written. What's the issue with that?

Reply:
I see